src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java

```diff
@@ -2,148 +2,73 @@
-        @Nullable final ItemStack output = inv.getItemInSlot(OUTPUT_SLOT);
-        if (input == null || input.getType() == Material.AIR
-            || !SlimefunUtils.isItemSimilar(input, SlimefunItems.OIL_BUCKET, true)
-            || (output != null
-            && (output.getType() != Items.RUBBER.getType()
-            || output.getAmount() == output.getMaxStackSize()
-            || !Items.RUBBER.getItem().isItem(output)))
-        ) return;
-
-        final BlockPosition pos = new BlockPosition(b.getWorld(), b.getX(), b.getY(), b.getZ());
-        int currentProgress = progress.getOrDefault(pos, -1);
-
-        // Process first tick - remove an input and put it in map.
-        if (currentProgress == -1 && takePower(b)) {
-            progress.put(pos, 0);
-            return;
-        }
-
-        // No progress and no input item, no tick needed. Or if there was no power (but can be processed)
-        if (currentProgress == -1 || !takePower(b)) return;
+    @Override
+    public List<ItemStack> getDisplayRecipes() {
+        List<ItemStack> displayRecipes = new ArrayList<>(recipes.size() * 2);
 
-        if (currentProgress == PROGRESS_AMOUNT) {
-            inv.consumeItem(INPUT_SLOT);
-            if (output != null && output.getAmount() > 0)
-                output.setAmount(output.getAmount() + 1);
-            else {
-                inv.replaceExistingItem(OUTPUT_SLOT, Items.RUBBER.clone());
-            }
-            progress.remove(pos);
-            ChestMenuUtils.updateProgressbar(inv, PROGRESS_SLOT, PROGRESS_AMOUNT, PROGRESS_AMOUNT, progressItem);
-        } else {
-            progress.put(pos, ++currentProgress);
-            ChestMenuUtils.updateProgressbar(inv, PROGRESS_SLOT, PROGRESS_AMOUNT - currentProgress, PROGRESS_AMOUNT,
-                progressItem);
+        for (MachineRecipe recipe : recipes) {
+            displayRecipes.add(recipe.getInput()[0]);
+            displayRecipes.add(recipe.getOutput()[recipe.getOutput().length - 1]);
         }
-    }
 
-    private boolean takePower(@Nonnull Block b) {
-        if (ChargableBlock.getCharge(b) < ENERGY_CONSUMPTION) return false;
-        ChargableBlock.addCharge(b, -ENERGY_CONSUMPTION);
-        return true;
+        return displayRecipes;
     }
 
     @Override
-    public EnergyNetComponentType getEnergyComponentType() {
-        return EnergyNetComponentType.CONSUMER;
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.FIRE_CHARGE);
     }
 
     @Override
-    public int getCapacity() {
-        return CAPACITY;
+    public String getInventoryTitle() {
+        return "&6Rubber Synthesizer";
     }
 
     @Override
-    public int[] getInputSlots() {
-        return new int[] {INPUT_SLOT};
+    public String getMachineIdentifier() {
+        return "RUBBER_SYNTHESIZER";
     }
 
     @Override
-    public int[] getOutputSlots() {
-        return new int[] {OUTPUT_SLOT};
+    public int getCapacity() {
+        return 256;
```

Comment:
is this the same as before?

Reply:
Defined it in a variable above but forgot to set